pages/index.js

```diff
@@ -182,20 +227,98 @@ export default function Home() {
   const [selectedMetric, setSelectedMetric] = useState()
   const displayedMetric = selectedMetric || metrics?.[0]
 
+  useEffect(() => {
+    // Add a listener to hide the tooltip when the user clicks outside of the tooltip contents.
+    const onClick = (event) => {
+      const tooltipEl = document.querySelector('#chart-tooltip')
+      if (tooltipEl && !tooltipEl?.contains(event.target)) {
+        tooltipEl.style.visibility = 'hidden'
+      }
+    }
+    window.addEventListener('click', onClick)
+    return () => {
+      window.removeEventListener('click', onClick)
+    }
+  }, [])
+
   return (
-    <Layout>
-      {!metrics && <div className='w-full h-full flex flex-col items-center justify-center text-lg'>Loading...</div>}
-      {metrics && !metrics.length && (
-        <div className='w-full flex flex-col items-center justify-center text-lg'>No data available.</div>
-      )}
-      {metrics && metrics.length && (
-        <dl className='mb-4 grid grid-cols-1 gap-5 sm:grid-cols-3'>
-          {metrics.map((metric) => (
-            <MetricCard key={metric.id} metric={metric} onSelect={() => setSelectedMetric(metric)} />
-          ))}
-        </dl>
-      )}
-      {displayedMetric && <Metric metric={displayedMetric || metrics?.[0]} />}
-    </Layout>
+    <>
+      <ZoomPlugin />
+      <Layout>
+        {!metrics && <div className='w-full h-full flex flex-col items-center justify-center text-lg'>Loading...</div>}
+        {metrics && !metrics.length && (
+          <div className='w-full flex flex-col items-center justify-center text-lg'>No data available.</div>
+        )}
+        {!!metrics?.length && (
+          <dl className='mb-4 grid grid-cols-1 gap-5 sm:grid-cols-3'>
+            {metrics.map((metric) => (
+              <MetricCard key={metric.id} metric={metric} onSelect={() => setSelectedMetric(metric)} />
+            ))}
+          </dl>
+        )}
+        {displayedMetric && <Metric metric={displayedMetric || metrics?.[0]} />}
+      </Layout>
+    </>
   )
 }
+
+// We only create once the tooltip and update it when needed.
+const getOrCreateTooltip = (chart) => {
+  let tooltipEl = chart.canvas.parentNode.querySelector('#chart-tooltip')
```

Comment:
Why is this rendered outside of React, can't we just render it in React, and have a regular onClick handler in the chart?

Reply:
Updated in: https://github.com/youknowriad/codehealth/pull/1/commits/475be98cf350b085a95a03f17c212dd9cebf61e3